google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkServiceManager.java

```diff
@@ -43,6 +77,196 @@ public void onNewCloudSdkServiceTypeSelected(CloudSdkServiceType newServiceType)
     }
   }
 
+  /**
+   * Waits in background for Cloud SDK to be ready for all operations and then runs the given
+   * runnable. If process results in error or user cancel, shows notification and does not run. This
+   * method does not block and must be called from application UI thread.
+   *
+   * @param project Project to which runnable belongs.
+   * @param runAfterSdkReady Runnable to execute after Cloud SDK is ready. This runnable will be
+   *     executed on the application UI thread.
+   * @param progressMessage Message to show in progress dialog to identify which process is
+   *     starting, i.e. deployment or local server.
+   * @param sdkLogger Logger for errors etc.
+   */
+  public void runWhenSdkReady(
+      @Nullable Project project,
+      @NotNull Runnable runAfterSdkReady,
```

Comment:
nit: usually start the callback name with something like `after`, `on`, `before` etc. So in this case, something like `afterSdkReady`